### PR TITLE
Rest api docs

### PR DESCRIPTION
**Pull request by AndyTWF**
> _NOTE TO REVIEWERS_ - please **do not review PRs in the `DRAFT` state**, as the PR may change substantially before it is ready to review. Thanks.

## Description

Updates the REST API docs to:

- Mark event name as optional for a message publish
- Include the status code in the return section for message publish

## Review

Instructions on how to review the PR. 

Page to review: docs/api/rest-api#publish

### CODE REVIEW

File: content/api/rest-api.textile [hunk 1]
```diff
@@ -322,8 +322,8 @@ Publish a message on a channel. Note that since the REST API is stateless, publi
 The request body contains message details and is an object of the form:
 
 bc[json]. {
-  name: <event name>,
   data: <message payload>,
+  name: <optional event name>,
```
Comment (m-hulbert) [inline content/api/rest-api.textile:326]: Hey @AndyTWF, 

Are `name` and `data` not both technically optional going by [RSL1e](https://sdk.ably.com/builds/ably/specification/main/features/#RSL1e) (and a quick test)?

Comment (AndyTWF) [reply content/api/rest-api.textile:326]: Yes technically, I shall update that!

Comment (AndyTWF) [reply content/api/rest-api.textile:326]: Done in f325ab04ac730e747cfec996d0bda82a80283db812bee4559b4cbe6a6dcaca9d

**Review by m-hulbert — APPROVED** [review]
LGTM!

**Pull request merged**